Sources/SwiftSyntax/generated/syntaxNodes/SyntaxNodesAB.swift

```diff
@@ -207,6 +207,21 @@ public struct AccessorBlockSyntax: SyntaxProtocol, SyntaxHashable {
           \Self.unexpectedAfterRightBrace
         ])
   }
+  
+  @available(*, deprecated, message: "This cast will always fail")
+  public func `is`<S: SyntaxProtocol>(_ syntaxType: S.Type) -> Bool {
+    return self.as(syntaxType) != nil
+  }
+  
+  @available(*, deprecated, message: "This cast will always fail")
+  public func `as`<S: SyntaxProtocol>(_ syntaxType: S.Type) -> S? {
+    return S.init(self)
+  }
+  
+  @available(*, deprecated, message: "This cast will always fail")
+  public func cast<S: SyntaxProtocol>(_ syntaxType: S.Type) -> S {
+    return self.as(S.self)!
+  }
```

Comment:
I think we should be able to avoid generating all these methods by creating the following protocol and then conforming all the non-base nodes to it.

```swift
/// Protocol that syntax nodes conform to which don’t have any semantic subtypes, i.e. syntax nodes that aren't base nodes.
public protocol LeafSyntaxNodeProtocol: SyntaxProtocol {}

public extension LeafSyntaxNodeProtocol {
  @available(*, deprecated, message: "This cast will always fail")
  public func `is`<S: SyntaxProtocol>(_ syntaxType: S.Type) -> Bool {
    return self.as(syntaxType) != nil
  }

  @available(*, deprecated, message: "This cast will always fail")
  public func `as`<S: SyntaxProtocol>(_ syntaxType: S.Type) -> S? {
    return S.init(self)
  }

  @available(*, deprecated, message: "This cast will always fail")
  public func cast<S: SyntaxProtocol>(_ syntaxType: S.Type) -> S {
    return self.as(S.self)!
  }
}
```

Reply:
Oh yeah, I believe we should be able to do that. Thanks!

Reply:
> This cast will always fail

Isn't *necessarily* true right 😅? They could be casting to the same type. I suspect that's usually not the case though, so probably makes sense still.

Reply:
In our codebase, there are a few instances where we're casting leaf nodes to their own types. I've assumed this is redundant and unnecessary. Have I overlooked something?

Reply:
Oh, I've just realized you probably meant the content of the message, not the mechanism. 😄 I need coffee 🙈

Reply:
Sorry, to be clear I was referring to:
> "This cast will always fail"

In that case it won't fail, it's just pointless. The message is fine, I was just being pedantic :)

Reply:
Introduced 🎸

Reply:
Good comment, @bnbarham. 

I think we should introduce the `as`/`is`/`cast` methods as follows on `SyntaxProtocol` that will catch same-type conversions an emit a warning the the cast will always succeed, rather than fail.

```swift
  @available(*, deprecated, message: "This cast will always succeed")
  func `as`(_ syntaxType: Self.Type) -> Self? {
    return self
  }
```

Reply:
Really good idea! Implemented in https://github.com/apple/swift-syntax/pull/2108/commits/9986d57dfd80800c30b686ae05b55f8739c168e9